File: testng/src/main/java/com/shoppingapp/CartService.java
package com.shoppingapp;

import java.util.HashMap;
import java.util.Map;

public class CartService {
    private Map<String, Cart> carts = new HashMap<>();

    public Cart createCart(String userId) {
        Cart cart = new Cart();
        carts.put(userId, cart);
        return cart;
    }

    public Cart getCart(String userId) {
        return carts.get(userId);
    }

    public void checkoutCart(String userId) {
        // Checkout logic
        carts.remove(userId);
    }
}

File: behave/src/main/shoppingapp/cart_service.py
from src.main.shoppingapp.cart import Cart
class CartService:
    def __init__(self):
        self.carts = {}

    def create_cart(self, user_id):
        cart = Cart()
        self.carts[user_id] = cart
        return cart

    def get_cart(self, user_id):
        return self.carts.get(user_id)

    def checkout_cart(self, user_id):
        self.carts.pop(user_id, None)
